presale/models/presale_order.py

```diff
@@ -0,0 +1,89 @@
+from odoo import api, fields, models
+
+
+class PresaleOrder(models.Model):
+    _name = 'presale.order'
+    _description = 'presales orders'
+
+    name = fields.Char(string='Preorder Reference',
+                       required=True, copy=False, readonly=True,
+                       default=lambda self: 'New')
+
+    active = fields.Boolean('Active', default=True)
+
+    state = fields.Selection(
+        selection=[
+            ('draft', 'Draft'),
+            ('confirmed', 'Confirmed')
+        ],
+        string='Status',
+        readonly=True, copy=False,
+        default='draft'
+    )
+
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string="Customer",
+        required=True, readonly=False, change_default=True)
+
+    order_lines = fields.One2many(
+        'presale.order.line', 'order_id', string='Order Lines')
+
+    sale_order_id = fields.Many2one(
+        'sale.order', string='Sale Order', copy=False)
+
+    def validate_action(self):
+        return True
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get('name', "New") == "New":
+                vals['name'] = self.env['ir.sequence'].next_by_code(
+                    'presale.order') or "New"
+
+        return super().create(vals_list)
+
+    def validate_action(self):
```

Comment:
This method can possibly called on multiple records at once, this would lead to errors or incoherences because it is writing the same values to all records in `self`. You can circumvent this by using `for order in self:` and working with `order`, or by adding `self.ensure_one()` at the start of your method if the method should never be called on multiple records